08.ls_object/file_info.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class FileInfo
+  FILE_TYPE = {
+    'file' => '-',
+    'directory' => 'd',
+    'link' => 'l',
+    'socket' => 's'
+  }.freeze
+
+  FILE_PERMISSION = {
+    '7' => 'rwx',
+    '6' => 'rw-',
+    '5' => 'r-x',
+    '4' => 'r--',
+    '3' => '-wx',
+    '2' => '-w-',
+    '1' => '--x',
+    '0' => '---'
+  }.freeze
+
+  def initialize(file)
+    @file_name = file
+    @file_stat = File.stat(file)
+  end
+
+  def filename
+    @file_name
+  end
+
+  def type
+    FILE_TYPE[@file_stat.ftype]
+  end
+
+  def permission
+    permission = @file_stat.mode.to_s(8).chars
+    permission_integer = [permission[-3], permission[-2], permission[-1]]
+    permission_rwx = permission_integer.map do |number|
+      FILE_PERMISSION[number]
+    end
+    permission_rwx.join
```

Comment:
```suggestion
    permission_integer.map { |number| FILE_PERMISSION[number] }.join
```

のようにメソッドチェーンを使う方法もあります。

Reply:
メソッドチェーンについて知らなかったので、勉強になりました。
確かにコードも短くなって読みやすくなった気がします👀

メソッドチェーンを使った書き方に修正しました！